activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -239,9 +239,18 @@ def as_json(options = nil)
   end
 end
 
-class IPAddr # :nodoc:
-  def as_json(options = nil)
-    to_s
+unless defined?(IPAddr::VERSION) && Gem::Version.new(IPAddr::VERSION) >= Gem::Version.new("1.2.7")
```

Comment:
If use ipaddr gem v1.2.7+, use native `IPAddr#as_json`

Note for myself : `defined?(IPAddr::VERSION)` is necessary because `IPAddr::VERSION` is available since v1.2.4 via https://github.com/ruby/ipaddr/pull/33 and Ruby 3.1.0 ships ipaddr v1.2.3 as a default gem.

* Ruby 3.1.0 Released
https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/

> ipaddr 1.2.3

```
$ ruby -v
ruby 3.1.0p0 (2021-12-25 revision fb4df44d16) [x86_64-linux]
$ gem info ipaddr

*** LOCAL GEMS ***

ipaddr (1.2.3)
    Authors: Akinori MUSHA, Hajimu UMEMOTO
    Homepage: https://github.com/ruby/ipaddr
    Licenses: Ruby, BSD-2-Clause
    Installed at (default): /home/yahonda/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0

    A class to manipulate an IP address in ruby
$ irb
irb(main):001:0> require 'ipaddr'
=> true
irb(main):002:0> IPAddr::VERSION
(irb):2:in `<main>': uninitialized constant IPAddr::VERSION (NameError)
	from /home/yahonda/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /home/yahonda/.rbenv/versions/3.1.0/bin/irb:25:in `load'
	from /home/yahonda/.rbenv/versions/3.1.0/bin/irb:25:in `<main>'
irb(main):003:0>
```